src/forms/NumberInput.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import cx from 'classnames';
+import Flex from '../layout/Flex';
+import FlexItem from '../layout/FlexItem';
+import Icon from '../media/Icon';
+
+export const DECREMENT_BTN_CLASS = 'decrementButton';
+export const INCREMENT_BTN_CLASS = 'incrementButton';
+
+/**
+ * @module NumberInput
+ */
+class NumberInput extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+
+		this.decrement = this.decrement.bind(this);
+		this.increment = this.increment.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onFocus = this.onFocus.bind(this);
+	}
+
+	onBlur(e) {
+		if (document.activeElement !== this.fauxInputEl && document.activeElement !== this.decrementBtnEl && document.activeElement !== this.incrementBtnEl) {
+			this.fauxInputEl.classList.remove('focused');
+		}
+	}
+
+	onChange(e) {
+		this.setState({ value: e.target.value });
+		if (this.props.onChange) {
+			this.props.onChange(e);
+		}
+	}
+
+	onFocus(e) {
+		this.fauxInputEl.classList.add('focused');
+	}
+
+	increment() {
+		const currentVal = new Number(this.state.value);
+		const increment = new Number(this.props.step);
+		const newValue = currentVal + increment;
+
+		if (newValue > this.props.max) return;
```

Comment:
If someone is incrementing and it's over the max, should the new value be max then?